src/NextjsDistribution.ts

```diff
@@ -175,32 +181,38 @@ export class NextjsDistribution extends Construct {
     return this.props.functionUrlAuthType === lambda.FunctionUrlAuthType.AWS_IAM;
   }
 
-  private createStaticBehaviorOptions(): cloudfront.BehaviorOptions {
-    const responseHeadersPolicy = new ResponseHeadersPolicy(this, 'StaticResponseHeadersPolicy', {
-      // add default header for static assets
-      customHeadersBehavior: {
-        customHeaders: [
-          {
-            header: 'cache-control',
-            override: false,
-            // MDN Cache-Control Use Case: Caching static assets with "cache busting"
-            // @see: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#caching_static_assets_with_cache_busting
-            value: `max-age=${Duration.days(365).toSeconds()}, immutable`,
-          },
-        ],
-      },
-      securityHeadersBehavior: this.commonSecurityHeadersBehavior,
-      comment: 'Nextjs Static Response Headers Policy',
-      ...this.props.overrides?.staticResponseHeadersPolicyProps,
-    });
+  private createStaticBehaviorOptions(): BehaviorOptions {
+    const staticBehaviorOptions = this.props.overrides?.staticBehaviorOptions;
+
+    // create default response headers policy if not provided
+    const responseHeadersPolicy =
+      staticBehaviorOptions?.responseHeadersPolicy ??
+      new ResponseHeadersPolicy(this, 'StaticResponseHeadersPolicy', {
+        // add default header for static assets
+        customHeadersBehavior: {
+          customHeaders: [
+            {
+              header: 'cache-control',
+              override: false,
+              // MDN Cache-Control Use Case: Caching static assets with "cache busting"
+              // @see: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#caching_static_assets_with_cache_busting
+              value: `max-age=${Duration.days(365).toSeconds()}, immutable`,
+            },
+          ],
+        },
+        securityHeadersBehavior: this.commonSecurityHeadersBehavior,
+        comment: 'Nextjs Static Response Headers Policy',
+        ...this.props.overrides?.staticResponseHeadersPolicyProps,
+      });
+
     return {
       ...this.commonBehaviorOptions,
       origin: this.s3Origin,
       allowedMethods: cloudfront.AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
       cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD_OPTIONS,
       cachePolicy: cloudfront.CachePolicy.CACHING_OPTIMIZED,
+      ...staticBehaviorOptions,
```

Comment:
could we keep `...staticBehaviorOptions` at bottom of this object? I think overrides should always be last object to spread.

Reply:
I know this won't change the behavior of construct, but it's more for the principle of always having overrides as last object to spread.

Reply:
that's what I was saying that overrides overwrites overrides, if I pass `responseHeadersPolicy: null`, it will create a default behavior but then when this `... staticBehaviorOptions` happens it will overwrite the created behavior with null, because only `undefined` does not overwrites, leading to errors.

Reply:
Is `null` allowed? I thought only the value or `undefined`?

Reply:
in js all is possible :), on the users's code they can pass null if they are not enforcing this check on ts transpile.

Reply:
or, a variable could be null for some reason, but you sill think you are passing it, and in this case we as lib should make sure we have a default for that.

Reply:
![2024-01-05T10-44-07](https://github.com/jetbridge/cdk-nextjs/assets/980905/c335b9f3-0d5a-4c16-9540-0ab12b24f683)


btw, let me refresh my understanding of this, but on the browser at least even explicit undefined will overwrite it, it will only not overwrite if the property is not passed.

Reply:
yeap, that's it, if you pass `responseHeadersPolicy: null | undefined` explicitly (different than not passing the property) the undefined value will overwrite the created policy.

Reply:
I think we can assume proper types are being followed. I think you're right though that we should make sure the policies are last in object because `undefined` will cause this same issue :/

Reply:
Sorry to go back and forth on this again, but if that's the desired behavior of user? What if they don't want a cache policy or response headers policy to be defined? If we keep overrides spread at bottom, then we give them the power to do this. Otherwise, we take this option away from them.

Reply:
yeah, it makes sense